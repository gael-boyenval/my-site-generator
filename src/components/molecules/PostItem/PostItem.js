import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Link from 'gatsby-link';
import { Title3, TinyHeading } from 'Atoms/Typography';
import { rem } from 'Styles/utils';
import { unit, colors, spaces } from 'Styles/tokens';

const Wrapper = styled.div`
  display: flex;
  flex-flow: column nowrap;
  min-height: ${rem(unit * 16)};
`;

const Metas = styled.div`
  padding-bottom: ${rem(unit)};
  color: ${colors.base.light};
  border-bottom: solid 1px ${colors.base.light};
`;

const Title = styled.div`
  flex: 100%;
  display: flex;
  padding-top: ${rem(spaces.m)};
`;

class PostItem extends PureComponent {
  static propTypes = {
    post: PropTypes.shape({
      fields: PropTypes.shape({
        slug: PropTypes.string.isRequired,
      }).isRequired,
      frontmatter: PropTypes.shape({
        date: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  render() {
    const { post } = this.props;
    const { fields, frontmatter } = post;

    return (
      <Wrapper key={fields.slug}>
        <Metas>
          <TinyHeading>{frontmatter.date}</TinyHeading>
        </Metas>
        <Title>
          <Link to={`${fields.slug}`}>
            <Title3>{frontmatter.title}</Title3>
          </Link>
        </Title>
      </Wrapper>
    );
  }
}

export default PostItem;
