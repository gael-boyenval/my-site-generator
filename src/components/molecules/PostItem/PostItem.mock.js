export default {
  fields: {
    slug: 'cest-un-hash-d-url',
  },
  frontmatter: {
    date: '2018-05-03',
    title: 'câ€™est un super titre',
  },
};
