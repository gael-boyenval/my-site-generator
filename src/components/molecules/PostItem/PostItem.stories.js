/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

import PostItem from './PostItem';
import mock from './PostItem.mock';

storiesOf('03 - Molecules', module)
  .addDecorator(withKnobs)
  .add('PostItem', withInfo()(() => <PostItem post={mock} />));
