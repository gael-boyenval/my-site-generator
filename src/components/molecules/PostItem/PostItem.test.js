/* eslint-disable no-undef */
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';

import PostItem from './PostItem';
import mock from './PostItem.mock';

const shallowWithRouter = (node) =>
  shallow(<Router initialEntries={[{ pathname: '/', key: 'testKey' }]}>{node}</Router>);

describe('Component PostItem', () => {
  it('should be render without crash', () => {
    const component = shallowWithRouter(<PostItem post={mock} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
