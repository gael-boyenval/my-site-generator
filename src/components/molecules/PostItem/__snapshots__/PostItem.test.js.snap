// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PostItem should be render without crash 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PostItem
    post={
      Object {
        "fields": Object {
          "slug": "cest-un-hash-d-url",
        },
        "frontmatter": Object {
          "date": "2018-05-03",
          "title": "câ€™est un super titre",
        },
      }
    }
  />
</Router>
`;
